src/main/java/io/stargate/sgv2/jsonapi/service/bridge/executor/CollectionSettings.java

```diff
@@ -0,0 +1,123 @@
+package io.stargate.sgv2.jsonapi.service.bridge.executor;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.stargate.bridge.proto.QueryOuterClass;
+import io.stargate.bridge.proto.Schema;
+import io.stargate.sgv2.jsonapi.config.constants.DocumentConstants;
+import io.stargate.sgv2.jsonapi.exception.ErrorCode;
+import io.stargate.sgv2.jsonapi.exception.JsonApiException;
+import java.util.Optional;
+
+/**
+ * Refactored as seperate class that represent a collection property.
+ *
+ * @param collectionName
+ * @param vectorEnabled
+ * @param vectorSize
+ * @param similarityFunction
+ * @param vectorizeServiceName
+ * @param modelName
+ */
+public record CollectionSettings(
+    String collectionName,
+    Boolean vectorEnabled,
```

Comment:
Can this be "unknown"? If not (which I think is the case), should perhaps be primitive `boolean` instead? Unless the reason is to allow exclusion of `null` Boolean value on serialization.